platform/mellanox/cmis_host_mgmt/cmis_host_mgmt.py

```diff
@@ -0,0 +1,151 @@
+        }
+    }
+
+    @staticmethod
+    def change_param(param, path, action):
+        file_path = '{}/{}'.format(path, CMISHostMgmtActivator.PARAMS[param]["file_name"])
+        lines = None
+
+        with open(file_path, 'r') as param_file:
+            lines = param_file.read()
+
+        if lines:
+            if action == "disable":
+                lines = re.sub(CMISHostMgmtActivator.PARAMS[param]["enabled_param"],
+                               CMISHostMgmtActivator.PARAMS[param]["disabled_param"],
+                               lines)
+            elif action == "enable":
+                if param == "sai_profile" and not re.search(CMISHostMgmtActivator.PARAMS[param]["disabled_param"], lines):
+                    with open(file_path, 'a') as param_file:
+                        param_file.write(CMISHostMgmtActivator.PARAMS[param]["enabled_param"])
+                    return
+
+                lines = re.sub(CMISHostMgmtActivator.PARAMS[param]["disabled_param"],
+                               CMISHostMgmtActivator.PARAMS[param]["enabled_param"],
+                               lines)
+
+        with open(file_path, 'w') as param_file:
+            param_file.write(lines)
+
+
+    @staticmethod
+    def parse_show_platform_summary():
+        summary = subprocess.check_output(['show', 'platform', 'summary'])
+        summary = summary.decode('utf-8')
+        summary = [x for x in summary.split('\n') if x]
+
+        for field in summary:
+            key, value = field.split(": ")
+            
+            if key == 'Platform':
+                platform = value
+
+            elif key == 'HwSKU':
+                sku = value
+
+        return platform, sku
+
+
+    @staticmethod
+    def remove_im_file(file_path):
+        if os.path.isfile(file_path):
+            os.remove(file_path)
+
+
+    @staticmethod
+    def copy_im_file(src_path, dest_path):
+        if os.path.isfile(src_path):
+            shutil.copy(src_path, dest_path)
+
+
+    @staticmethod
+    def disable_im():
+        platform, sku = CMISHostMgmtActivator.parse_show_platform_summary()
+        sku_path = '/usr/share/sonic/device/{0}/{1}'.format(platform, sku)
+        platform_path = '/usr/share/sonic/device/{0}'.format(platform)
+        CMISHostMgmtActivator.change_param("sai_profile", sku_path, 'disable')
+        CMISHostMgmtActivator.change_param("pmon_daemon_control", platform_path, 'disable')
+        
+        CMISHostMgmtActivator.remove_im_file('{0}/{1}'.format(sku_path, 'media_settings.json'))
+        CMISHostMgmtActivator.remove_im_file('{0}/{1}'.format(sku_path,'optics_si_settings.json'))
+        CMISHostMgmtActivator.remove_im_file('{0}/{1}'.format(platform_path, 'media_settings.json'))
+        CMISHostMgmtActivator.remove_im_file('{0}/{1}'.format(platform_path, 'optics_si_settings.json'))
+        CMISHostMgmtActivator.remove_im_file('{0}/{1}'.format(sku_path, 'pmon_daemon_control.json'))
```

Comment:
what will happen if pmon_daemon_control.json will be only under SKU folder? 
and then user wants to disable the feature and you will remove it? 
I would suggest to check - if there is such file under platform folder, remove the file under SKU and change the platform one param. if there is no such file, change param in the file under SKU

Reply:
Fixed